File: Removal of String.cpp
#include <iostream>
#include <string>
#include <unordered_set>
#include <algorithm>
using namespace std;

string removeDuplicatesAndReverse(const string& s)
{
    string result;
    unordered_set<char> seen;
    for (auto it = s.rbegin(); it != s.rend(); ++it) 
    {
        char c = *it;
        if (seen.count(c) == 0) 
        {
            result.push_back(c);
            seen.insert(c);
        }
    }
    return result;
}

string applyOperations(const string& s)
{
    string current = s;
    while (true) 
    {
        unordered_set<char> seen;
        bool hasDuplicates = false;
        for (char c : current) 
        {
            if (seen.count(c) > 0) 
            {
                hasDuplicates = true;
                break;
            }
            seen.insert(c);
        }
        if (hasDuplicates) 
        {
            current = removeDuplicatesAndReverse(current);
        }
        else 
        {
            break;
        }
    }
    return current;                             
}

int main()
{
   string s;
    cin >> s;
    string result = applyOperations(s);
    cout << result << endl;
    return 0;
}

File: Removal of String.py
def remove_duplicates_and_reverse(s):
    result = []
    seen = set()
    for c in reversed(s):
        if c not in seen:
            result.append(c)
            seen.add(c)
    return ''.join(result)


def apply_operations(s):
    current = s
    while True:
        seen = set()
        has_duplicates = False
        for c in current:
            if c in seen:
                has_duplicates = True
                break
            seen.add(c)
        if has_duplicates:
            current = remove_duplicates_and_reverse(current)
        else:
            break
    return current


def main():
    import sys
    input = sys.stdin.read
    s = input().strip()
    result = apply_operations(s)
    print(result)


if __name__ == "__main__":
    main()
